File: eulerground-python/5.py
#Smallest positive number evenly divisible by 1 to 20
from sys import argv
def gcd(a,b):
    a,b = max(a,b),min(a,b)
    if a % b == 0: return b
    else: return gcd(b,a%b)

def lcm(a,b):
    return a*b/gcd(a,b)

def main(n):
    res = 1
    for i in range(1,n+1):
        res = lcm(res,i)
    print res
if __name__ == '__main__':
    if len(argv) == 2:
        main(eval(argv[1]))
    elif len(argv) > 2 :
        print "Usage : python 5.py <upper natural number limit>"
    else:
        main(20)

File: eulerground-c++/5.cpp
//Smallest positive evenly divisible by all from 1 to 20
#include <iostream>
#include <cstdio>
using namespace std;

long long gcd(long long a,long long b)
{
    if(a<b){
        a=a+b;
        b=b-a;
        a=a-b;
        }
    if (a%b == 0)
        return b;
    else return a%b;
}

long long lcm (long long a, long long b)
{
    return (a*b)/gcd(a,b);

}
int main()
{
    long long res = 1;
    int n,i;
    cout <<"Upper limit: ";
    cin >> n;
    for(i=1;i<=n;i++)res=lcm(res,i);
    cout<<res;
    return 0;
}
